cmd/generatetf/generator/generator.go

```diff
@@ -162,10 +162,16 @@ func sourceName(source client.Source) string {
 }
 
 // generateDestination generates a destination resource block from an API destination object and a terraform destination type and ConfigMeta.
-func generateDestination(destination client.Destination, terraformType string, cm *configs.ConfigMeta) (*hclwrite.Block, error) {
+func generateDestination(destination client.Destination, terraformType string, cm *configs.ConfigMeta) (block *hclwrite.Block, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic while generating destination block: %v", r)
+		}
+	}()
```

Comment:
Do we need a separate section like this in generateSource() as well ?

Reply:
Good point yes. Sources are not requiring this kind of generation yet because they don't have any config, but it's good to have this guardrail there as well